fetch-jwk.go

```diff
@@ -40,7 +42,12 @@ func FromIssuerClaim() func(*jwt.Token) (interface{}, error) {
 			return nil, err
 		}
 
-		return getKey(keySet, keyID)
+		key, err := getKey(keySet, keyID)
+		if err == errKeyNotFound {
+			delete(issuerCache, issuer)
+			return FromIssuerClaim()(token)
```

Comment:
We need some way to stop after first retry, since if the issuer also doesn't have this key this will possibly create an infinite recursion